<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>987</width>
    <height>432</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_6">
    <item row="0" column="0">
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="title">
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QSplitter" name="splitter_21">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QTextBrowser" name="textBrowser">
           <property name="html">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Здравствуйте, это приложение для составления простых заданий на понимание рандоминизированных дерамид поиска. Это приложение состоит из 2-х видов заданий. Когда будете выполнять задания, следует тщательно записывать ответ ибо вернуться к заданию будет &lt;/span&gt;&lt;span style=&quot; font-size:9pt; font-weight:600; text-decoration: underline;&quot;&gt;нельзя!&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;Первое задание.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;    1)Чтобы написать ЛПК(лево, право, ключ) - сначала идем влево (если такой указатель существет), затем идем вправо (если такой указатель существет), затем записываем ключ.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;    2)Чтобы написать ЛКП(лево, ключ, право) - сначала идем влево (если такой указатель существет), затем записываем ключ, затем идем вправо (если такой указатель существет).&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;    3)Чтобы написать КЛП(ключ, лево, право) - сначала записываем ключ, затем идем влево (если такой указатель существет), затем идем вправо (если такой указатель существет).&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:9pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;Второе задание.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;    В данное дерево вставляется один элемент с фиксированным приоритетом и рандомным значением. Этот элемент вставляется по следующим правилам:&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;    1)Если элемент, на котором мы находимся, нулевой, то присваеваем ему наш вставляемый элемент. &lt;br /&gt;    2)Если приоритет элемента меньше приоритета элемента, на котором мы находимся, то элемент вставляется по правилу 3, иначе по правилу 4.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;    3)Если ключ вставляемого элемента больше элемента, накотором мы находимся, то идем в правый указатель, иначе в левый.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;    4)Если ключ вставляемого элемента больше элемента, накотором мы находимся, то идем в левый указатель, иначе в правый.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;    &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;    Требуется написать:&lt;br /&gt;    1)ЛПК после вставки элемента.&lt;br /&gt;    2)ЛКП после вставки элемента.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;    3)КЛП после вставки элемента.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
          <widget class="QPushButton" name="pushButton">
           <property name="text">
            <string>Начать тест</string>
           </property>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="the_task_1">
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QSplitter" name="splitter">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QLabel" name="LRK">
           <property name="text">
            <string>Введите через пробел ЛПК дерева. (пробелы должны быть только между ключами)</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="lineEditLRK"/>
         </widget>
        </item>
        <item row="0" column="1" rowspan="4">
         <widget class="QGraphicsView" name="graphicsView">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="sizeIncrement">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAsNeeded</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAsNeeded</enum>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QAbstractScrollArea::AdjustIgnored</enum>
          </property>
          <property name="dragMode">
           <enum>QGraphicsView::ScrollHandDrag</enum>
          </property>
          <property name="transformationAnchor">
           <enum>QGraphicsView::AnchorViewCenter</enum>
          </property>
          <property name="resizeAnchor">
           <enum>QGraphicsView::AnchorViewCenter</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QSplitter" name="splitter_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QLabel" name="KLR">
           <property name="text">
            <string>Введите через пробел КЛП дерева. (пробелы должны быть только между ключами)</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="lineEditKLR"/>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QSplitter" name="splitter_7">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QLabel" name="FinishResults">
           <property name="text">
            <string/>
           </property>
          </widget>
          <widget class="QPushButton" name="FINISH_THE_TASK">
           <property name="text">
            <string>Следующее задание</string>
           </property>
          </widget>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QSplitter" name="splitter_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QLabel" name="LKR">
           <property name="text">
            <string>Введите через пробел ЛКП дерева. (пробелы должны быть только между ключами)</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="lineEditLKR"/>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_2">
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="QLabel" name="FinishResults_2">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="0" column="1" rowspan="5">
         <widget class="QGraphicsView" name="graphicsView_3">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="sizeIncrement">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAsNeeded</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAsNeeded</enum>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QAbstractScrollArea::AdjustIgnored</enum>
          </property>
          <property name="dragMode">
           <enum>QGraphicsView::ScrollHandDrag</enum>
          </property>
          <property name="transformationAnchor">
           <enum>QGraphicsView::AnchorViewCenter</enum>
          </property>
          <property name="resizeAnchor">
           <enum>QGraphicsView::AnchorViewCenter</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QSplitter" name="splitter_8">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QLabel" name="LRK_4">
           <property name="text">
            <string>Вставлен ключ. Введите через пробел ЛПК дерева с изменениями. (пробелы должны быть только между ключами)</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="lineEditLRK_4"/>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QSplitter" name="splitter_10">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QLabel" name="LKR_4">
           <property name="text">
            <string>Вставлен ключ. Введите через пробел ЛКП дерева с изменениями. (пробелы должны быть только между ключами)</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="lineEditLKR_4"/>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QSplitter" name="splitter_9">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QLabel" name="KLR_4">
           <property name="text">
            <string>Вставлен ключ. Введите через пробел КЛП дерева с изменениями. (пробелы должны быть только между ключами)</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="lineEditKLR_4"/>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QPushButton" name="FINISH_THE_TASK_3">
          <property name="text">
           <string>Следующее задание</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="results">
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <widget class="QSplitter" name="splitter_22">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QSplitter" name="splitter_20">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <widget class="QTextBrowser" name="AllResults"/>
           <widget class="QTextBrowser" name="AllResults_2"/>
          </widget>
          <widget class="QPushButton" name="pushButton_2">
           <property name="text">
            <string>Закрыть приложение</string>
           </property>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>987</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuTreapDemonstratio">
    <property name="title">
     <string>Project</string>
    </property>
    <addaction name="separator"/>
    <addaction name="Open"/>
    <addaction name="Enter"/>
    <addaction name="actionCreate_variant"/>
    <addaction name="actionDelete_treap"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuTreap">
    <property name="title">
     <string>Treap</string>
    </property>
    <addaction name="actionEnter_Key"/>
    <addaction name="actionErase_Key"/>
   </widget>
   <addaction name="menuTreapDemonstratio"/>
   <addaction name="menuTreap"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="Open">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="Enter">
   <property name="text">
    <string>Enter</string>
   </property>
  </action>
  <action name="actionEnter_Key">
   <property name="text">
    <string>Enter Key</string>
   </property>
  </action>
  <action name="actionErase_Key">
   <property name="text">
    <string>Erase Key</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionStep_by_step_presentation">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Step by step presentation</string>
   </property>
  </action>
  <action name="actionCreate_variant">
   <property name="text">
    <string>Create variant</string>
   </property>
  </action>
  <action name="actionDelete_Key">
   <property name="text">
    <string>Delete Key</string>
   </property>
  </action>
  <action name="actionDelete_treap">
   <property name="text">
    <string>Delete treap</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
